ika/client.go

```diff
@@ -78,3 +85,96 @@ func (p *Client) UpdateLC(
 		RequestType: "WaitForLocalExecution",
 	})
 }
+
+// ApproveAndSign approves and signs a set of messages using the IKA network. Returns its signatures
+func (p *Client) ApproveAndSign(
+	ctx context.Context,
+	dwalletCapID string,
+	signMessagesID string,
+	messages [][]byte,
+	logger zerolog.Logger,
+) ([][]byte, error) {
+
+	//TODO: This function was only tested against dummy implementation of the dwallet module deployed locally.
+	//Once it is ready, test it again
+	req := models.MoveCallRequest{
+		Signer:          p.Signer.Address,
+		PackageObjectId: p.LcPackage,
+		Module:          p.Module,
+		Function:        "approve_messages",
+		TypeArguments:   []interface{}{},
+		Arguments: []interface{}{
+			dwalletCapID,
+			messages,
+		},
+		GasBudget: p.GasBudget,
+	}
+	messageApprovals, err := p.c.MoveCall(ctx, req)
+	if err != nil {
+		logger.Err(err).Msg("Error calling move function:")
+		return nil, err
+	}
+
+	req = models.MoveCallRequest{
+		Signer:          p.Signer.Address,
+		PackageObjectId: p.LcPackage,
+		Module:          p.Module,
+		Function:        "sign",
+		TypeArguments:   []interface{}{},
+		Arguments: []interface{}{
+			signMessagesID,
+			messageApprovals,
+		},
+		GasBudget: p.GasBudget,
+	}
+	resp, err := p.c.MoveCall(ctx, req)
+	if err != nil {
+		logger.Err(err).Msg("Error calling move function:")
+		return nil, err
```

Comment:
to avoid double logging, error logging should be higher up. You can wrap errors if we need to add additional information. See https://go.dev/blog/go1.13-errors

Note: In contrast to errors, Info and Debug logs should be at the place where the event occurs.